torchmdnet/module.py

```diff
@@ -163,64 +198,68 @@ def optimizer_step(self, *args, **kwargs):
         super().optimizer_step(*args, **kwargs)
         optimizer.zero_grad()
 
-    def training_epoch_end(self, training_step_outputs):
-        dm = self.trainer.datamodule
-        if hasattr(dm, "test_dataset") and len(dm.test_dataset) > 0:
-            should_reset = (
-                self.current_epoch % self.hparams.test_interval == 0
-                or (self.current_epoch + 1) % self.hparams.test_interval == 0
-            )
-            if should_reset:
-                # reset validation dataloaders before and after testing epoch, which is faster
-                # than skipping test validation steps by returning None
-                self.trainer.reset_val_dataloader(self)
+    def _get_mean_loss_dict_for_type(self, type):
+        # Returns a list with the mean loss for each loss_fn for each stage (train, val, test)
+        # Parameters:
+        # type: either y, neg_dy or total
+        # Returns:
+        # A dict with an entry for each stage (train, val, test) with the mean loss for each loss_fn (e.g. mse_loss)
+        # The key for each entry is "stage_type_loss_fn"
+        assert self.losses is not None
+        self.mean_losses = {}
```

Comment:
why `self.`? I don't see this used anywhere else

Reply:
You are right!